src/test/perl/test-caffileeditor_source.t

```diff
@@ -0,0 +1,77 @@
+#!/usr/bin/perl
+use strict;
+use warnings;
+use FindBin qw($Bin);
+use lib "$Bin/modules";
+use CAF::FileEditor;
+use Test::More tests => 7;
+use Test::MockModule;
+use Test::Quattor::Object;
+use Carp qw(confess);
+use File::Path;
+use File::Temp qw(tempfile);
+use Readonly;
+
+$SIG{__DIE__} = \&confess;
+
+# FIXME:
+# LC::File::file_contents (used by CAF::FileEditor constructor) doesn't work 
+# in the unit test context. Mock it until we find a better solution...
+our $lcfile = Test::MockModule->new("LC::File");
+
+sub read_file_contents {
+    my $fname = shift;
+    my $fh;
+    open($fh, "<", $fname) || die ("failed to open $fname");
+    my $contents = join('', <$fh>);
+    return $contents;
+}
+$lcfile->mock("file_contents", \&read_file_contents);
+
+
+my $testdir = 'target/test/editor';
+mkpath($testdir);
+(undef, my $filename) = tempfile(DIR => $testdir);
+
+Readonly my $TEXT => <<EOF;
+En un lugar de La Mancha, de cuyo nombre no quiero acordarme
+no ha tiempo que vivía un hidalgo de los de lanza en astillero...
+EOF
+Readonly my $ANOTHER_TEXT => "adarga antigua, rocín flaco y galgo corredor.";
+
+my $fh;
+my $obj = Test::Quattor::Object->new();
+
+
+# Create a file and check that it is empty
+($fh, $filename) = tempfile(DIR => $testdir);
+$fh->close();
+$fh = CAF::FileEditor->new($filename, log => $obj);
+is("$fh","","Existing file ($filename) empty");
+$fh->close();
+
+# Check that reference file contents is used as the initial contents when it
+# is newer than the file edited.
+sleep 2;
+my ($ref_fh, $ref_filename) = tempfile(DIR => $testdir);
+print $ref_fh $TEXT;
+$ref_fh->close();
+$fh = CAF::FileEditor->new($filename, log => $obj, source => $ref_filename);
+is(*$fh->{options}->{source},$ref_filename,"File source is correctly defined");
+ok($fh->_is_reference_newer(),"Source file ($ref_filename) is newer than actual file ($filename)");
+is("$fh",$TEXT,"Reference file ($filename) contents used");
+$fh->close();
+
+# Check that reference file contents is not used as the initial contents when it
+# is older than the file edited.
+sleep 2;
+my ($new_fh, $new_filename) = tempfile(DIR => $testdir);
+print $new_fh $ANOTHER_TEXT;
+$new_fh->close();
+$new_fh = CAF::FileEditor->new($new_filename, log => $obj, source => $ref_filename);
+is(*$new_fh->{options}->{source},$ref_filename,"File source is correctly defined");
```

Comment:
I don't have a strong opinion. But I don't think we tend to delete options and this may make the troubleshooting more difficult... This is the role of the constructor to decide what to do. If it is really important to know if the source was used, I'd prefer to add an attribute rather than remove the option....